docs/engineering/android-app/configuring/config-types/questionnaire.mdx

```diff
@@ -325,3 +326,43 @@ triggerConditions | This defines an array of condition for to be met for the eve
 eventResourceId |  uniqueId of resource id to be closed |  yes   |               |
 eventResources | A list of resources to close(Type of ResourceConfig) |    yes           |     |      |
 
+
+## Unique ID assignment
+
+Unique ID configs are used to configure how generated unique ID is extracted from a FHIR resource. Currently the Group Resource
+is used to sync generated identifiers. 
+
+Here is as sample configuration for the unique identifier assignment:
+
+```json
+{
+  "uniqueIdAssignment": {
+    "linkId": "phn",
+    "idFhirPathExpression": "Group.characteristic.where(exclude=false and code.text='phn').first().value.text",
```

Comment:
Can you explain why this should be configurable? I think you did a great job writing the code to work based on how this is configured, but that leads to some performance compromises

Reply:
1. The first assumption was we'd not expect multiple items in Group#characteristics
2. This was configured to support other ResourceTypes in the future; not to limit the functionality to Group resources. Extracting data from resources using FHIR path was also an easier alternaitve compared to hard-coding implementation.